Components/VariantConfigurator.php

```diff
@@ -57,9 +57,11 @@ public function configureVariantAttributes(Product $product, Detail $detail)
             $configSet = new Set();
             $configSet->setName('Set-' . $article->getName());
             $configSet->setArticles([$article]);
+            $configSet->setType($product->configuratorSetType);
             $article->setConfiguratorSet($configSet);
         } else {
             $configSet = $article->getConfiguratorSet();
+            $configSet->setType($product->configuratorSetType);
```

Comment:
I didn't see any changes in Product verificator in SDK. So ```$product->configuratorSetType``` can be null even when $product->variant is not empty. I'd say that you need to verify if his value is > 0. Also please prepare another PR in SDK where you explain what different values mean. Because 1,2,3 is not clear for me. For customers which are using systems different from shopware will be the same. Please add some example  in dochint of $product::configuratorSetType